### PR TITLE
Check for duplicated sysctl keys

### PR DESCRIPTION
**Pull request by mmirecki**
The tuning-cni config currently allows to specif duplicated sysctls, also with conflicting values. Duplicated sysctl entries are usually  not used intentionally, but are rather the result of an error in configuration.
This PR adds a check for duplicated sysctl's.

### CODE REVIEW

**Comment by squeed** [comment]
@mmirecki looks good! Can you fix DCO and go fmt? Then I can merge when CI goes green.

File: plugins/meta/tuning/tuning.go [hunk 1]
```diff
@@ -542,3 +545,27 @@ func readAllowlist() (bool, []string, error) {
...
+	if _, exists := sysctlDuplicatesMap[key]; exists {
+		return errors.New("duplicated sysctl keys are not allowed")
+	}
+	sysctlDuplicatesMap[key] = ""
+	return nil
+}
+
+func validateSysctlConflictingValues(data []byte) error {
+	// Need to reset values for use in tests
+	//sysctlDuplicatesMap = map[sysctlKey]interface{}{}
```
Comment (squeed) [inline plugins/meta/tuning/tuning.go:568]: Did you mean to leave this commented out?

Comment (mmirecki) [reply plugins/meta/tuning/tuning.go]: That's taken care of in the tests. Fixed.
Thanks

File: plugins/meta/tuning/tuning.go [hunk 2]
```diff
@@ -542,3 +545,25 @@ func readAllowlist() (bool, []string, error) {
...
+func (d *sysctlKey) UnmarshalText(data []byte) error {
+	key := sysctlKey(string(data))
+	if _, exists := sysctlDuplicatesMap[key]; exists {
+		return errors.New("duplicated sysctl keys are not allowed")
+	}
+	sysctlDuplicatesMap[key] = ""
+	return nil
+}
+
+func validateSysctlConflictingValues(data []byte) error {
```
Comment (mars1024) [inline plugins/meta/tuning/tuning.go:566]: Actually this function is checking conflicting keys not values, i think it should be renamed?

Comment (mmirecki) [reply plugins/meta/tuning/tuning.go]: indeed, fixed

**Comment by squeed** [comment]
@mmirecki @dcbw observes we should do this when parsing the config for CHECK too, right?

File: plugins/meta/tuning/tuning.go (hunk 1 above)
Comment (mmirecki) [reply plugins/meta/tuning/tuning.go]: > @mmirecki @dcbw observes we should do this when parsing the config for CHECK too, right?

@squeed @dcbw Done

**Review by mccv1r0 — APPROVED** [review]

**Pull request merged**